public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -140,21 +154,33 @@ export class DashboardGrid extends PureComponent<Props> {
         panelPos.isDraggable = panel.collapsed;
       }
 
+      this.eventSubs.add(panel.events.subscribe(PanelGridPositionLayoutEvent, this.onUpdatePanelGridPositionLayout));
       layout.push(panelPos);
     }
 
     return layout;
   }
 
+  onUpdatePanelGridPositionLayout = (event: PanelGridPositionLayoutEvent) => {
+    const indexGridItem = this.state.layout.findIndex((item) => Number.parseInt(item.i, 10) === event.payload.panelId);
+    const newLayout = [...this.state.layout];
+    newLayout[indexGridItem] = { ...newLayout[indexGridItem], ...event.payload.gridPos };
+    this.setState({ layout: newLayout, shouldForceUpdate: false });
+    this.updateGridPos(newLayout[indexGridItem], newLayout);
+  };
+
   onLayoutChange = (newLayout: ReactGridLayout.Layout[]) => {
     for (const newPos of newLayout) {
       this.panelMap[newPos.i!].updateGridPos(newPos);
     }
 
     this.props.dashboard.sortPanelsByGridPos();
 
-    // Call render() after any changes.  This is called when the layout loads
-    this.forceUpdate();
+    // Call render() after any changes.  This is called when the initial layout loads
+    // After layout is updated via setState
+    if (this.state.shouldForceUpdate) {
+      this.forceUpdate();
+    }
```

Comment:
Out of curiosity, this changes the intent of the previous implementation as it would always force render previously. I'm just asking if this is an intended change and if you can explain the reason?

Reply:
Indeed 😄 ,  before was needed because the `layout` property was not part of the props or the state, and the component was not being re-render when the user was changing the panel sizes.

On my current implementation when I added the `layout` to the state it was re-rendering twice (First, doing `setState` of layout, and a second time when the `onLayoutChange` was called.

Still, I had to keep this `forceUpdate` for the initial render, when the component is mounted and the layout loads.

Reply:
Ok, thanks for a great explanation 👍